### PR TITLE
Repo lifecycle additions (Partial Delivery, Reprice, Adjustment, Substitution) 

### PR DESCRIPTION
**Pull request by davidalk**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,12 +1,52 @@
...
+
+_Partial Delivery
+
+In the repo market it is a recommended best practice that if a seller fails to deliver the full amount of collateral
+on settlement date, that the buyer agrees to accept the amount of collateral delivered and agree on the remaining amount and when
+it will be delivered.
+
+_What is being released?_
+
+A new function `Create_PartialDeliveryPrimitivenstruction` is added that accepts inputs of the delivered amount and undelivered amount.
```
Comment (chrisisla) [inline RELEASE.md:15]: ...Primitive**I**nstruction

File: RELEASE.md [hunk 2]
```diff
@@ -1,12 +1,52 @@
...
+_What is being released?_
+
+A new function `Create_PartialDeliveryPrimitivenstruction` is added that accepts inputs of the delivered amount and undelivered amount. 
+The function closes the existing trade and opens two new trades, one based on the delivered quantity and one based on the undelivered
+quantity. The function returns a primitive instruction that can be used to execute a business event. 
+There is a also a new qualification function `Qualify_PartialDelivery` that validates the results meet the partial delivery criteria.
+
+_Reprice and Adjustment
+
+Repricing and Adjustment are methods used to elimintate net exposures between counterparties. In both methods the original transaction is terminated
```
Comment (chrisisla) [inline RELEASE.md:22]: eliminate

File: RELEASE.md [hunk 3]
```diff
@@ -1,12 +1,52 @@
...
+In the repo market, if agreed by the counterparties on the contract, it is possible for the seller to replace collateral that was originally
+delivered at settlement with different collateral. The new collateral will most likely have a new quantity and price. 
+
+_What is being released?_
 
-The changes can be reviewed in PR [#2460](https://github.com/finos/common-domain-model/pull/2460).
+A new function `Create_SubstitutionPrimitiveInstruction` is added that accepts inputs of a new payout that include new cash and collateral
+amounts and price, and returns a new primitive instruction that can be used to execute a business event. In the case where
+there was a contractual agreement to substitute collateral the original trade remains with reference to it's contract, identifiers, and 
+other details. There is a also a new qualification function `Qualify_Substitution` that validates the results meet the substitution criteria.
```
Comment (chrisisla) [inline RELEASE.md:52]: At the bottom just need to add a link to a PR number

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/partial-delivery.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+### Create Shaping Business Event
```
Comment (chrisisla) [inline rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/partial-delivery.md:1]: Is this title correct?

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-adjustment.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+### Create Repo Reprice Business Event
```
Comment (chrisisla) [inline rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-adjustment.md:1]: Is the detail in this .md correct? Are we using the reprice as an example of an adjustment?

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-substitution.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+### Create Repo Reprice Business Event
```
Comment (chrisisla) [inline rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-substitution.md:1]: Are the contents of this .md file correct for a substitution? The title looks incorrect but the content looks okay...

File: RELEASE.md (hunk 1 above)
Comment (tomhealey-icma) [reply RELEASE.md:15]: agreed

File: RELEASE.md (hunk 2 above)
Comment (tomhealey-icma) [reply RELEASE.md:22]: agreed

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/partial-delivery.md (hunk 4 above)
Comment (tomhealey-icma) [reply rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/partial-delivery.md:1]: No. Needs to be corrected.

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-substitution.md (hunk 6 above)
Comment (tomhealey-icma) [reply rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-substitution.md:1]: The md files includes all four events.

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/partial-delivery.md (hunk 4 above)
Comment (tomhealey-icma) [reply rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/partial-delivery.md:1]: fixed

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-adjustment.md (hunk 5 above)
Comment (tomhealey-icma) [reply rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-adjustment.md:1]: fixed

File: rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-substitution.md (hunk 6 above)
Comment (tomhealey-icma) [reply rosetta-source/src/main/resources/cdm-sample-files/functions/repo-and-bond/repo-substitution.md:1]: fixed

**Comment by tomhealey-icma** [comment]
This is a copy of the PR I submitted, [PR2472](https://github.com/finos/common-domain-model/pull/2472). Please review and comment on the original. I've submitted a new commit with changes.

**Pull request closed without merging**